shell/platform/android/vsync_waiter_android.cc

```diff
@@ -43,11 +38,28 @@ void VsyncWaiterAndroid::AwaitVSync() {
   });
 }
 
-static void OnNativeVsync(JNIEnv* env,
-                          jclass jcaller,
-                          jlong frameTimeNanos,
-                          jlong frameTargetTimeNanos,
-                          jlong java_baton) {
+float VsyncWaiterAndroid::GetDisplayRefreshRate() const {
+  JNIEnv* env = fml::jni::AttachCurrentThread();
+  if (g_vsync_waiter_class == nullptr) {
+    return kUnknownRefreshRateFPS;
+  }
+  jclass clazz = g_vsync_waiter_class->obj();
+  if (clazz == nullptr) {
+    return kUnknownRefreshRateFPS;
+  }
+  jfieldID fid = env->GetStaticFieldID(clazz, "refreshRateFPS", "F");
+  // We can safely read this 32-bit float from Java in any thread because
+  // 32-bits read and write are guaranteed to be atomic:
+  // https://stackoverflow.com/questions/11459543/should-getters-and-setters-be-synchronized/11459616#11459616
```

Comment:
I'd remove this URL.  Atomicity of float reads is defined in the Java language spec.

Reply:
I didn't add that link. Just moved that code around.

Reply:
I'll get rid of it anyway.